packages/graphql-connection-transformer/src/resources.ts

```diff
@@ -189,4 +193,214 @@ export class ResourceFactory {
             )
         }).dependsOn(ResourceConstants.RESOURCES.GraphQLSchemaLogicalID)
     }
+
+    // Resources for new way to parameterize @connection
+
+    /**
+     * Create a get item resolver for singular connections.
+     * @param type The parent type name.
+     * @param field The connection field name.
+     * @param relatedType The name of the related type to fetch from.
+     * @param connectionAttributes The names of the underlying attributes containing the fields to query by.
+     * @param keySchema Key schema of the index or table being queried.
+     */
+    public makeGetItemNewConnectionResolver(type: string,
+                                           field: string,
+                                           relatedType: string,
+                                           connectionAttributes: string[],
+                                           keySchema: KeySchema[]): Resolver {
+
+        const partitionKeyName = keySchema[0].AttributeName as string
+
+        let keyObj : ObjectNode = obj({
+            [partitionKeyName] :
+                ref(`util.dynamodb.toDynamoDBJson($util.defaultIfNullOrBlank($ctx.source.${connectionAttributes[0]}, "${NONE_VALUE}"))`)
+            })
+
+        // Add sort key if there is one.
+        if (connectionAttributes.length > 2) {
+            const rangeKeyFields = connectionAttributes.slice(1);
+            const sortKeyName = keySchema[1].AttributeName as string
+            const condensedSortKeyValue = this.condenseRangeKey(
+                rangeKeyFields.map(keyField => `\${ctx.args.${keyField}}`)
```

Comment:
Won't these values come off the `$ctx.source`? Can we make sure there is a test case that tests this GetOne scenario.

Reply:
You're right there was a missing test and error here, fixed now.